src/core/db/query_rewriter.py

```diff
@@ -0,0 +1,336 @@
+                if table[0][2] is not None:
+                    query = '(SELECT * FROM %s.%s.%s' % (
+                        table[0][2], table[0][0], table[0][1])
+                else:
+                    query = '(SELECT * FROM %s.%s' % (table[0][0], table[0][1])
+
+                policies = self.find_security_policy(table[0][1],
+                                                     table[0][0],
+                                                     "select",
+                                                     table[0][2])
+                if policies:
+                    query += ' WHERE '
+                    for policy in policies:
+                        query += policy + " OR "
+                    query = query[:-4]
+                query += ")"
+
+                # Here we are handling table aliasing. In the case where the
+                # default query does not use an alias, we need to auto-create
+                # an alias for the table (since we are constructing a subquery
+                # from the table name to apply row level security). We then
+                # need to replace all later instances of the original table
+                # with the alias.
+                if table[1] != "":
+                    query += " %s" % table[1]
+                else:
+                    original_table_name = table[0][0] + "." + table[0][1]
+                    alias_name = table[0][0] + table[0][1]
+                    query += " AS %s" % (alias_name)
+                    replace_list.append((original_table_name,
+                                        alias_name,
+                                        len(result) + len(query)))
+
+                result += query
+                if len(table_information) > 1:
+                    result += ", "
+
+            if len(table_information) > 1:
+                result = result[:-2]
+
+            prev_token = token
+
+        for alias in replace_list:
+            result = result[0:alias[2]] + result[alias[2]:].replace(
+                alias[0] + ".", alias[1] + ".")
+            result = result[0:alias[2]] + result[alias[2]:].replace(
+                alias[0] + " ", alias[1] + " ")
+
+        result = result.replace("USERNAME", "'" + self.user + "'")
+        return result
+
+    def find_security_policy(self, table, repo, policytype, repo_base):
+        '''
+        Look up policies associated with the table and repo and returns a
+        list of all the policies defined for the user.
+        '''
+
+        if repo_base is None:
+            repo_base = self.repo_base
+
+        with core.db.rlsmanager.RowLevelSecurityManager(
+            username=self.user,
+            repo_base=repo_base,
+            repo=repo,
+            table=table
+        ) as mngr:
+            user_policies = mngr.find_security_policy(
+                policy_type=policytype, grantee=self.user)
+            all_policies = mngr.find_security_policy(
```

Comment:
@kxzhang - there's a special case implied here where the `grantee = 'ALL'`. But as far as I can tell, `rlsmanager.find_security_policy` and `pg.find_security_policy()` never implement this. Is 'ALL' supposed to be a wildcard? And in this case, wouldn't it make sense just to leave out the `grantee` argument entirely? 

Or are users allowed to define a grantee as `'ALL'`? In this case, we should document that feature, and make sure that datahub users can not ever be named `'ALL'`